Cargo.toml

```diff
@@ -10,7 +10,7 @@ repository = "https://github.com/mmtk/mmtk-core"
 readme = "README.md"
 categories = ["memory-management"]
 keywords = ["gc", "garbage", "collection", "garbage-collection", "allocation"]
-rust-version = "1.70.0"
+rust-version = "1.71.1"
```

Comment:
This breaks our MSRV policy. Our pinned Rust version is `1.71.1` and the MSRV should be at max `1.70`. We can think about bumping our pinned version (to 1.77).

Reply:
I bumped `rust-toolchain` to 1.77.0.